software/winrm/src/main/java/org/apache/brooklyn/location/winrm/WinRmMachineLocation.java

```diff
@@ -388,20 +401,38 @@ public static String getDefaultUserMetadataString() {
                 "}"
         ));
 
+        // FIXME USE_SECURE_WINRM
+        // Missing generate certificate step.
+        // https://support.microsoft.com/en-us/kb/2019527
+        //
+        // One possible approach is to generate a certificate and append it to this command.
+        // http://stackoverflow.com/questions/1615871/creating-an-x509-certificate-in-java-without-bouncycastle
```

Comment:
The certificate is best generated on the machine, without leaving it ever (for self-signed case). On the other hand it's not possible to get the public part at this step. 

I see this setup step dissappearing longer term (or keeping it minimalistic). Instead do something like the jclouds init sequence where it connects with whatever is provided by the cloud, then configuring it to our liking.